WeakAurasOptions/OptionsFrames/OptionsFrame.lua

```diff
@@ -898,13 +898,16 @@ function OptionsPrivate.CreateFrame()
   dynamicTextCodesFrame:Hide()
 
   function OptionsPrivate.ToggleTextReplacements(data, widget, event)
-    -- This throttle prevents double toggling as a result from OnEditFocusLost
+    -- If the text edit has focus when the user clicks on the button, we'll get two events:
+    -- a) The OnEditFocusLost
+    -- b) The ToggleButton OnClick event
+    -- Since we want to hide the text replacement window in that case,
+    -- ignore the ToggleButton if it is directly after the  OnEditFocusLost
```

Comment:
The code will be read and even modified multiple times in the future, so if we introduce hacks like this we to should ensure that whoever touches the code in the future can understand which case lead to the hack.

Also the hack should be as specific as possible, which means I changed to exactly check for OnEditFocusLost -> ToggleButton.

Looking at the time difference between the events, they were pretty close to 0.1 on my machine, 0.1 didn't feel safe enough.